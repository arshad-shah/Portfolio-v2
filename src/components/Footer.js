import { Box, IconButton, Typography } from "@material-ui/core";
import React from "react";
import useStyles from "./Footer.styles";

// icons
import GitHubIcon from "@material-ui/icons/GitHub";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import MailIcon from "@material-ui/icons/Mail";

export default function Footer() {
  const classes = useStyles();
  return (
    <Box className={classes.root}>
      <Typography variant="h6">
        &copy; {new Date().getFullYear()} Designed and Created By: Arshad Shah
      </Typography>
      <Box>
        <IconButton
          role="button"
          className={classes.socialLinks}
          color="default"
          aria-label="link to Github account of Arshad shah"
          data-block="link to Github"
          href="https://github.com/arshad-shah"
        >
          <GitHubIcon />
        </IconButton>

        <IconButton
          role="button"
          className={classes.socialLinks}
          color="default"
          aria-label="link to Linkedin account of Arshad shah"
          data-block="link to Linkedin"
          href="https://www.linkedin.com/in/arshadshah"
        >
          <LinkedInIcon />
        </IconButton>

        <IconButton
          className={classes.socialLinks}
          color="default"
          role="button"
          aria-label="link to mail account of Arshad shah"
          data-block="link to Mail"
          href="mailto:shaharshad1999@outlook.com"
        >
          <MailIcon />
        </IconButton>
      </Box>
      <Typography variant="subtitle2">
        Google Play and the Google Play logo are trademarks of Google LLC.
      </Typography>
      <Typography variant="subtitle2">V2.0 Date: 15/August/2021</Typography>
    </Box>
  );
}
