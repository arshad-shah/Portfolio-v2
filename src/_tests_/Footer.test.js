import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from '../components/Footer';

describe('Footer of the site:', () => {
	it('should render footer with the creater name in it', () => {
		render(<Footer />);
		const creator = screen.getByText(/Arshad Shah/i);
		expect(creator).toBeInTheDocument();
	});

	it('should render the social account links', () => {
		render(<Footer />);
		const githubIcon = screen.getByRole('button', {
			name: /link to Github account of Arshad shah/i,
		});
		const linkedInIcon = screen.getByRole('button', {
			name: /link to Linkedin account of Arshad shah/i,
		});

		const hackerRankIcon = screen.getByRole('button', {
			name: /link to Hacker Rank account of Arshad shah/i,
		});
		const mailIcon = screen.getByRole('button', {
			name: /link to mail account of Arshad shah/i,
		});

		expect(githubIcon).toBeInTheDocument();
		expect(linkedInIcon).toBeInTheDocument();
		expect(hackerRankIcon).toBeInTheDocument();
		expect(mailIcon).toBeInTheDocument();
	});

	it('should render footer', () => {
		const { container } = render(<Footer />);
		expect(container).toMatchSnapshot();
	});
});
