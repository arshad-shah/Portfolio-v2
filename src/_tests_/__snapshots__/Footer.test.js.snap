// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer of the site: should render footer 1`] = `
<div>
  <div
    class="MuiBox-root css-lsidqv"
    data-testid="footer"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Â© 
      2022
       Arshad Shah
    </h6>
    <div
      class="MuiBox-root css-0"
    >
      <a
        aria-label="link to Github account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-nkneg1-MuiButtonBase-root-MuiIconButton-root"
        data-block="linktoGithub"
        data-mui-internal-clone-element="true"
        data-testid="githubLink"
        href="https://github.com/arshad-shah"
        role="button"
        tabindex="0"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        aria-label="link to Linkedin account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-nkneg1-MuiButtonBase-root-MuiIconButton-root"
        data-block="linktoLinkedin"
        data-mui-internal-clone-element="true"
        data-testid="linkedInLink"
        href="https://www.linkedin.com/in/arshadshah"
        role="button"
        tabindex="0"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
          />
          <rect
            height="12"
            width="4"
            x="2"
            y="9"
          />
          <circle
            cx="4"
            cy="4"
            r="2"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        aria-label="link to Hacker Rank account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-nkneg1-MuiButtonBase-root-MuiIconButton-root"
        data-block="linktoHackerRank"
        data-mui-internal-clone-element="true"
        data-testid="hackerRankLink"
        href="https://www.hackerrank.com/shaharshad57"
        role="button"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M477.5 128C463 103.05 285.13 0 256.16 0S49.25 102.79 34.84 128s-14.49 230.8 0 256 192.38 128 221.32 128S463 409.08 477.49 384s14.51-231 .01-256zM316.13 414.22c-4 0-40.91-35.77-38-38.69.87-.87 6.26-1.48 17.55-1.83 0-26.23.59-68.59.94-86.32 0-2-.44-3.43-.44-5.85h-79.93c0 7.1-.46 36.2 1.37 72.88.23 4.54-1.58 6-5.74 5.94-10.13 0-20.27-.11-30.41-.08-4.1 0-5.87-1.53-5.74-6.11.92-33.44 3-84-.15-212.67v-3.17c-9.67-.35-16.38-1-17.26-1.84-2.92-2.92 34.54-38.69 38.49-38.69s41.17 35.78 38.27 38.69c-.87.87-7.9 1.49-16.77 1.84v3.16c-2.42 25.75-2 79.59-2.63 105.39h80.26c0-4.55.39-34.74-1.2-83.64-.1-3.39.95-5.17 4.21-5.2 11.07-.08 22.15-.13 33.23-.06 3.46 0 4.57 1.72 4.5 5.38C333 354.64 336 341.29 336 373.69c8.87.35 16.82 1 17.69 1.84 2.88 2.91-33.62 38.69-37.58 38.69z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        aria-label="link to mail account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-nkneg1-MuiButtonBase-root-MuiIconButton-root"
        data-block="linktoMail"
        data-mui-internal-clone-element="true"
        data-testid="mailLink"
        href="mailto:arshad@arshadshah.com"
        role="button"
        tabindex="0"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
    >
      V3.2 - 01 / May / 2022
    </h6>
  </div>
</div>
`;
